.pre-commit-config.yaml

```diff
@@ -45,3 +44,36 @@ repos:
 #   rev: v1.3.0
 #   hooks:
 #     - id: dead
+- repo: https://github.com/jumanjihouse/pre-commit-hooks
+  rev: 1.11.0
+  hooks:
+    - id: script-must-have-extension
+      types: [shell, non-executable]
+    - id: shellcheck
+      args: [--color=always, --check-sourced, --external-sources]
+      additional_dependencies: [shellcheck]
+    - id: shfmt
+      additional_dependencies: [shfmt]
+      args: [-s, -sr, -w, -i, "2"]
+- repo: https://github.com/codespell-project/codespell.git
+  rev: v1.16.0
+  hooks:
+    - id: codespell
+      args: [--check-filenames, --check-hidden, --write-changes, --ignore-words-list, "iff"]
```

Comment:
Great typo finding tool. I fixed boat loads of typos with this. Just checks code changed in commit.